### PR TITLE
Documentation/plant structs

### PR DESCRIPTION
**Pull request by markus2330**
## Basics

- [ ] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.
- [ ] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Documentation is conforming to [our Documentation Guidelines](/doc/documentation.md)
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to our Coding Guidelines
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,7 +16,9 @@ cargo install --version 0.1.1 --git https://github.com/ElektraInitiative/mdbook-
 mdbook serve --open
 ```
 
-Note that Rust needs to be installed for it to work.
+Important links:
```
Comment (kitzbergerg) [inline README.md:19]: Maybe write a section documentation like in [Actix](https://github.com/actix/actix-web#documentation)?
Then it seems clearer what additional information there is (and we can add the code doc and mdbook websites as well once they are deployed)

File: doc/database/hierarchy.md [hunk 2]
```diff
@@ -116,6 +116,8 @@ We know about names (abstract and concrete, including hybrid):
 
 ## Attributes
 
+Columns are documented in [the struct](/backend/src/model/entity.rs).
```
Comment (kitzbergerg) [inline doc/database/hierarchy.md:119]:
```suggestion
Columns are documented in their [respective structs](/backend/src/model/entity.rs).
```
There are multiple structs in `entity.rs`.

File: README.md (hunk 1 above)
Comment (markus2330) [reply README.md:19]: I don't understand, it already is in a section Documentation? The self-building of mdbook is only as long as the CI does not do it, eventually in the section should be only links, like in Actix.

File: backend/src/model/entity.rs [hunk 3]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
     pub id: i32,
-    /// The unique name of the plant. The structure is described in the /doc/database/hierarchy.md document in the repository.
+
+    /// - The unique name of the plant.
+    /// - The structure is described above (`doc/database/hierarchy.md`).
+    /// - *Fill ratio:* 100%
     pub unique_name: String,
-    /// The list of the common names of the plant.
+
+    /// - The list of the common names of the plant in English.
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:39]:
```suggestion
    /// - common name of the plant in English.
```

File: backend/src/model/entity.rs [hunk 4]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+    /// - *Fetched from* PracticalPlants and Permapeople.
+    /// - *Fill ratio:* 90%
     pub common_name_en: Option<Vec<Option<String>>>,
-    /// The list of the common names of the plant in German. Fetched from Wikidata API if not present in the source datasets.
+
+    /// - The list of the common names of the plant in German.
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:44]:
```suggestion
    /// - common name of the plant in German.
```

File: backend/src/model/entity.rs [hunk 5]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+    /// - *Fetched from* Wikidata API if not present in any source datasets.
+    /// - *Fill ratio:* 25%
     pub common_name_de: Option<Vec<Option<String>>>,
-    /// Family of the plant.
+
+    /// - Family of the plant.
+    /// - See `/doc/architecture/glossary.md`.
+    /// - *Used* to build up hierarchy.
+    /// - *Fetched from* PracticalPlants and Permapeople.
+    /// - *Fill ratio:* 88%
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:53]: Family is determined by botanical nomenclature. Sometimes gets changed due to modern insights through genetical research.

File: backend/src/model/entity.rs [hunk 6]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
     pub family: Option<String>,
-    /// Genus of the plant.
+
+    /// - Genus of the plant.
+    /// - Is also first word of unique_name, see above (`/doc/database/hierarchy.md`)
+    /// - *See also* `/doc/architecture/glossary.md`.
+    /// - *Used* to build up hierarchy.
+    /// - *Fetched from* PracticalPlants and Permapeople.
+    /// - *TODO:* copy from first word of unique name.
+    /// - *Fill ratio:* 63%
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:62]: Genus is determined by botanical nomenclature. Sometimes gets changed due to modern insights through genetical research.

File: backend/src/model/entity.rs [hunk 7]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
     pub genus: Option<String>,
-    /// Fetched from Permapeople as \`edible_uses\` and merged with Reinsaat.  (fetched from Permapeople and Reinsaat)
+
+    /// - The edible use of the plant, answering: Which food can be produced from this plant?
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:65]:
```suggestion
    /// - The edible use of the plant, answering: Which food type can be produced from this plant (e.g. oil)?
```

File: backend/src/model/entity.rs [hunk 8]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+
+    /// - Only for references.
+    /// - *Fetched from* PracticalPlants)
+    /// - *Fill ratio:* 63%
     pub botanic: Option<String>,
-    /// (fetched from PracticalPlants)
+
+    /// - Only informational.
+    /// - *Fetched from* PracticalPlants
+    /// - *Fill ratio:* 1%
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:88]: Plants are not only used for food but also for other uses, e.g. fiber.

File: backend/src/model/entity.rs [hunk 9]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
     pub material_uses: Option<String>,
-    /// (fetched from PracticalPlants)
+
+    /// - *Used* for search ranking (diversity).
+    /// - *Fetched from* PracticalPlants)
+    /// - *Fill ratio:* 13%
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:93]: ecological funktion of the plant, e.g. nitrogen fixer

File: backend/src/model/entity.rs [hunk 10]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
     pub functions: Option<String>,
-    /// (fetched from PracticalPlants)
+
+    /// - Not used.
+    /// - *Use* hardiness_zone instead.
+    /// - *Fetched from* PracticalPlants
+    /// - *Fill ratio:* 0.05%
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:99]: indication of the heat range a plant endures

File: README.md (hunk 1 above)
Comment (kitzbergerg) [reply README.md:19]: Oh sorry. That's my bad. Github just cut out the Documentations section and I didn't realize. So all is good here.

File: backend/src/model/entity.rs [hunk 11]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
     pub heat_zone: Option<i16>,
-    /// (fetched from PracticalPlants)
+
+    /// - Shade tolerance of the plant, to be used together with light_requirement.
+    /// - *Used* in shade layer.
+    /// - *For example* a plant that has "no shade", should get a warning if placed in a shade.
+    /// - *Fetched from* PracticalPlants.
+    /// - *Fill ratio:* 63%
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:106]: No shade: full sun exposure
Ligth shade: moderately shaded throughout the day
Partial shade: about 3-6 hours of direct sunlight
Permanent shade: less than 3 hours of direct sunlight
Permanent deep shade: almost no sunlight/no direct sunlight

Comment (yvonnemarkl) [inline backend/src/model/entity.rs:106]: Shade indicates the shade tolerance. Plants can grow at better light conditions, however.
Warnings should only show if the a plant is moved into a too dark spot. No warning when moved into a lighter spot.

File: backend/src/model/entity.rs [hunk 12]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
     pub shade: Option<Shade>,
-    /// Merged between Permapeople and PracticalPlants. (fetched from PracticalPlants and Permapeople)
+
+    /// - *See* explanation in `/doc/architecture/context.md`
+    /// - *Used* in soil layer.
+    /// - *Fetched from* PracticalPlants and Permapeople (merged between Permapeople and PracticalPlants).
+    /// - *Fill ratio:* 1%
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:112]: soil ph can be tested by the user with simple means (e.g. litmus).

**Review by yvonnemarkl — CHANGES_REQUESTED** [review]

File: backend/src/model/entity.rs [hunk 13]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+    /// - Shade tolerance of the plant, to be used together with shade.
+    /// - *Used* in shade layer.
+    /// - *For example* a plant that has "Full sun", should get a warning if placed in a shade.
+    /// - **Fetched from*** PracticalPlants and Permapeople (merged with `sun` of PracticalPlants)
+    /// - *Fill ratio:* 88%
     pub light_requirement: Option<Vec<Option<LightRequirement>>>,
-    /// Merged with water of PracticalPlants. (fetched from PracticalPlants and Permapeople)
+
+    /// - *Used* in hydrology layer.
+    /// - *Fetched from* PracticalPlants and Permapeople (merged with `water` of PracticalPlants).
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:258]:
```suggestion
    /// - *Fetched from* PracticalPlants and Permapeople (merged with `water` of PracticalPlants).
    /// - water = completely aquatic; wet = drowned, (often) flooded or in general very moist, e.g. swamp; moist = humid, regular water supply, e.g. flat bed with humus; well drained = dry, little water input.
```

File: backend/src/model/entity.rs [hunk 14]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+    /// - *Fill ratio:* 88%
     pub water_requirement: Option<Vec<Option<WaterRequirement>>>,
-    /// (fetched from Permapeople)
+
+    /// - Only informational.
+    /// - *Fetched from* Permapeople (renamed from `propagation`)
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:263]:
```suggestion
    /// - *Fetched from* Permapeople (renamed from `propagation`)
    /// - How to reproduce a plant: cuttings = cut pieces of wood; layering = let low branches reach the soil to root; Seed - direct sow = sow directly the seeds; division = split the rhizomes (roots) into pieces; Spores = plant reproduces via spores (e.g. ferns, funghi); seed - transplant = raise indoors from seed and transplant to outdoors later
```

File: backend/src/model/entity.rs [hunk 15]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
-    /// (fetched from Permapeople)
+
+    /// - Not used.
+    /// - *Fetched from* Permapeople.
+    /// - *Fill ratio:* 36%
     pub introduced_into: Option<String>,
-    /// Fetched from Permapeople as \`layer\` and renamed. (fetched from Permapeople)
+
+    /// - Only informational.
+    /// - *Fetched from* Permapeople as \`layer\` and renamed.
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:300]:
```suggestion
    /// - *Fetched from* Permapeople as \`layer\` and renamed.
    /// - Habitus describes the shape of a plant.
```

File: backend/src/model/entity.rs [hunk 16]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
-    /// (fetched from Permapeople)
+
+    /// - Not used.
+    /// - *Fetched from* Permapeople.
+    /// - *Fill ratio:* 15%
     pub plants_of_the_world_online_link_synonym: Option<String>,
-    /// Fetched from PracticalPlants as \`pollinators\` and merged with \`pollination\` of Permapeople. (fetched from PracticalPlants and Permapeople)
+
+    /// - Only informational.
+    /// - *Fetched from* PracticalPlants as `pollinators` and merged with `pollination` of Permapeople.
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:330]:
```suggestion
    /// - *Fetched from* PracticalPlants as `pollinators` and merged with `pollination` of Permapeople.
    /// - Pollination is the process that the pollen (male part) gets united with the stamina (female part), e.g. via bees, wind.
```

File: backend/src/model/entity.rs [hunk 17]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+
+    /// - Not used.
+    /// - Nearly empty.
+    /// - *Fetched from* Permapeople.
+    /// - *Fill ratio:* 0.01%
     pub resistance: Option<String>,
-    /// (fetched from Permapeople)
+
+    /// - Only informational.
+    /// - *Fetched from* Permapeople.
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:346]:
```suggestion
    /// - *Fetched from* Permapeople.
    /// - Root type can be considered when planning polycultures, e.g. combining shallow roots with deep roots.
```

File: backend/src/model/entity.rs [hunk 18]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+    /// - *Fill ratio:* 0.24%
     pub root_type: Option<String>,
-    /// Fetched from Permapeople as \`seed_planting_depth\` and renamed. (fetched from Permapeople)
+
+    /// - Only informational.
+    /// - *Fetched from* Permapeople as `seed_planting_depth` and renamed.
+    /// - Reinsaat: `Sowing depth` should be copied to `seed_planting_depth_en`
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:352]:
```suggestion
    /// - Reinsaat: `Sowing depth` should be copied to `seed_planting_depth_en`
    /// - When sowing each plant has a specific value how deep the seeds should be covered with soil.
```

File: backend/src/model/entity.rs [hunk 19]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+    /// - *Fill ratio:* 0.07%
     pub seed_planting_depth_en: Option<String>,
-    /// (fetched from Permapeople)
+
+    /// - Only informational.
+    /// - *Fetched from* Permapeople.
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:357]:
```suggestion
    /// - *Fetched from* Permapeople.
    /// - expected average life span (in years) of a seed of a certain specie.
```

File: backend/src/model/entity.rs [hunk 20]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+
+    /// - Not used.
+    /// - *TODO:* ovlerap with `functions`.
+    /// - *Fetched from* Permapeople.
+    /// - *Fill ratio:* 2%
     pub utility: Option<String>,
-    /// (fetched from Permapeople)
+
+    /// - Important information.
+    /// - *Fetched from* Permapeople.
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:381]:
```suggestion
    /// - *Fetched from* Permapeople.
    /// - specific warnings for eather human, animal or environmental well-being, e.g. toxic, invasive.
```

File: backend/src/model/entity.rs [hunk 21]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+
+    /// - Only informational.
+    /// - *TODO:* ask how to interpret.
+    /// - *Fetched from* Permapeople.
+    /// - *Fill ratio:* 0.12%
     pub when_to_start_outdoors_weeks: Option<String>,
-    /// (fetched from Permapeople)
+
+    /// - Only informational.
+    /// - *Fetched from* Permapeople.
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:424]:
```suggestion
    /// - *Fetched from* Permapeople.
    /// - Stratification is the process that a seed must go through to get triggered for germination, e.g. by cold temperatures.
```

File: backend/src/model/entity.rs [hunk 22]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+    /// - *Fill ratio:* 0.03%
     pub cold_stratification_temperature: Option<String>,
-    /// (fetched from Permapeople)
+
+    /// - Only informational.
+    /// - *Fetched from* Permapeople.
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:429]:
```suggestion
    /// - *Fetched from* Permapeople.
    /// - Suggested change
    /// - *Fetched from* Permapeople.
    /// - Stratification is the process that a seed must go through to get triggered for germination, e.g. by a certain amount of time under cold temperatures.

```

File: backend/src/model/entity.rs [hunk 23]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
-    /// The external URL provided by the origin source.
+
+    /// - Not used.
+    /// - The external URL provided by the origin source.
+    /// - *Fill ratio:* 9%
     pub external_url: Option<String>,
-    /// Fetched from PracticalPlants as \`root_zone_tendency\` and merged with root_depth of Permapeople. (fetched from PracticalPlants and Permapeople)
+
+    /// - Only informational.
+    /// - *Fetched from* PracticalPlants as `root_zone_tendency` and merged with root_depth of Permapeople.
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:508]:
```suggestion
    /// - *Fetched from* PracticalPlants as `root_zone_tendency` and merged with root_depth of Permapeople.
    /// - Root depth can be considered when planning polycultures, e.g. combining shallow roots with deep roots.
```

File: backend/src/model/entity.rs [hunk 24]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+    /// - Only informational.
+    /// - *Fetched from* Reinsaat.
+    /// - `Required quantity of seeds` should be called `required_quantity_of_seeds_en`
+    /// - *Fill ratio:* 3%
     pub required_quantity_of_seeds_en: Option<String>,
-    /// (fetched from Reinsaat)
+
+    /// - Only informational.
+    /// - *Fetched from* Reinsaat.
+    /// - `Saattiefe` should be called `seed_planting_depth_de`
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:565]:
```suggestion
    /// - `Saattiefe` should be called `seed_planting_depth_de`
    /// - When sowing, each plant has a specific value how deep the seeds should be covered with soil.
```

File: backend/src/model/entity.rs [hunk 25]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+    /// - *Fill ratio:* 5%
     pub seed_planting_depth_de: Option<String>,
-    /// (fetched from Reinsaat)
+
+    /// - German version of housand grain weight (German: Tausendkornmasse)
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:569]:
```suggestion
    /// - German version of thousand grain weight (German: Tausendkornmasse)
```

File: backend/src/model/entity.rs [hunk 26]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+    /// - Only informational.
+    /// - *Fetched from* Reinsaat.
+    /// - Called `Tausendkornmasse` in Reinsaat
+    /// - *Fill ratio:* 4%
     pub seed_weight_1000_de: Option<String>,
-    /// (fetched from Reinsaat)
+
+    /// - English version of housand grain weight (German: Tausendkornmasse)
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:576]:
```suggestion
    /// - English version of thousand grain weight (German: Tausendkornmasse)
```

File: backend/src/model/entity.rs [hunk 27]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+    /// - The list of the common names of the plant in German.
+    /// - *Fetched from* Wikidata API if not present in any source datasets.
+    /// - *Fill ratio:* 25%
     pub common_name_de: Option<Vec<Option<String>>>,
-    /// Family of the plant.
+
+    /// - Family of the plant.
+    /// - See `/doc/architecture/glossary.md`.
+    /// - *Used* to build up hierarchy.
+    /// - *Fetched from* PracticalPlants and Permapeople.
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:52]:
```suggestion
    /// - *Fetched from* PracticalPlants and Permapeople.
    /// - Family is determined by botanical nomenclature. Sometimes gets changed due to modern insights through genetical research.
```

File: backend/src/model/entity.rs [hunk 28]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+    /// - *Fill ratio:* 88%
     pub family: Option<String>,
-    /// Genus of the plant.
+
+    /// - Genus of the plant.
+    /// - Is also first word of unique_name, see above (`/doc/database/hierarchy.md`)
+    /// - *See also* `/doc/architecture/glossary.md`.
+    /// - *Used* to build up hierarchy.
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:59]:
```suggestion
    /// - *Used* to build up hierarchy.
    /// - Genus is determined by botanical nomenclature. Sometimes gets changed due to modern insights through genetical research.
```

File: backend/src/model/entity.rs [hunk 29]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
     pub botanic: Option<String>,
-    /// (fetched from PracticalPlants)
+
+    /// - Only informational.
+    /// - *Fetched from* PracticalPlants
+    /// - *Fill ratio:* 1%
     pub material_uses: Option<String>,
-    /// (fetched from PracticalPlants)
+
+    /// - *Used* for search ranking (diversity).
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:91]:
```suggestion
    /// - *Used* for search ranking (diversity).
    /// - ecological and environmental function of the plant, e.g. nitrogen fixer
```

File: backend/src/model/entity.rs [hunk 30]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+    /// - *Fetched from* PracticalPlants)
+    /// - *Fill ratio:* 13%
     pub functions: Option<String>,
-    /// (fetched from PracticalPlants)
+
+    /// - Not used.
+    /// - *Use* hardiness_zone instead.
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:97]:
```suggestion
    /// - *Use* hardiness_zone instead.
    /// - indication of the heat range a plant endures.
```

File: backend/src/model/entity.rs [hunk 31]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+    /// - *Fetched from* PracticalPlants
+    /// - *Fill ratio:* 0.05%
     pub heat_zone: Option<i16>,
-    /// (fetched from PracticalPlants)
+
+    /// - Shade tolerance of the plant, to be used together with light_requirement.
+    /// - *Used* in shade layer.
+    /// - *For example* a plant that has "no shade", should get a warning if placed in a shade.
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:104]:
```suggestion
    /// - *For example* a plant that has "no shade", should get a warning if placed in a shade.
    /// - No shade: full sun exposure; Ligth shade: moderately shaded throughout the day; Partial shade: about 3-6 hours of direct sunlight; Permanent shade: less than 3 hours of direct sunlight; Permanent deep shade: almost no sunlight/no direct sunlight.
```

File: backend/src/model/entity.rs [hunk 32]
```diff
@@ -20,214 +20,583 @@ use super::r#enum::{
...
+    /// - *Fetched from* PracticalPlants.
```
Comment (yvonnemarkl) [inline backend/src/model/entity.rs:105]:
```suggestion
    /// - Shade indicates the shade tolerance. Plants can grow at better light conditions, however.
Warnings should only show if the a plant is moved into a too dark spot. No warning when moved into a lighter spot.
    /// - *Fetched from* PracticalPlants.
```

**Pull request merged**